portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -51,40 +54,50 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     for(Entity entity: results) {
       long id = entity.getKey().getId();
       String name = (String) entity.getProperty("name");
+      String email = (String) entity.getProperty("email");
       String message = (String) entity.getProperty("message");
       long timestamp = (long) entity.getProperty("timestamp");
 
-      Comment comment = new Comment(id, name, message, timestamp);
+      Comment comment = new Comment(id, name, email, message, timestamp);
       comments.add(comment);
     }
 
-    // Convert comments list to JSON
+    // Convert comments list to JSON.
     String json = GSON.toJson(comments);
 
-    // Send json to server
+    // Send json to server.
     response.setContentType("applications/json;");
     response.getWriter().println(json);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Check if the user is logged in and if not redirect them. If so get the email.
+    String email;
+    if(USER.isUserLoggedIn()) {
```

Comment:
if statement should be followed by a space. See https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace.